### ISSUE TITLE
[BUG] Node test fails when node uses named property from previous node output

### ISSUE THREAD

**Issue by bwilliams2** [microsoft/promptflow#1772]
**Bug Description**
Running node specific tests with `pf_client.test` fails with an `InvalidReferenceProperty` error when the tested node utilizes a property from a node output as a node input value (e.g., `${node.output.property}`). 

**How To Reproduce the bug**
An example of the bug is shown at https://github.com/bwilliams2/pf_node_test_example and abbreviated below.

Failing node definition in flow.dag.yaml 
https://github.com/bwilliams2/pf_node_test_example/blob/32446bef67783c7b65390aa1e18a21a2e06a705a/flow/flow.dag.yaml#L26-L32
```
- name: echo
  type: python
  source:
    type: code
    path: echo.py
  inputs:
    echo: ${echo_object.output.echo}
```

Failing `pf_client.test` call
https://github.com/bwilliams2/pf_node_test_example/blob/32446bef67783c7b65390aa1e18a21a2e06a705a/test_flow.py#L7
```
pf.test("flow/flow.dag.yaml", inputs={"echo": "Hello"}, node="echo")
```

Output of execution
```
pf.test("flow/flow.dag.yaml", inputs={"echo": "Hello"}, node="echo")
```

```
 File ".pyenv/versions/3.11.5/lib/python3.11/site-packages/promptflow/executor/_input_assignment_parser.py", line 105, in parse_node_property
    raise InvalidReferenceProperty(message_format=message_format, property=property, node_name=node_name) from e
promptflow.executor._errors.InvalidReferenceProperty: Flow execution failed. Invalid property 'echo' when accessing the node 'echo_object'. Please check the property and try again.
```

**Expected behavior**
It is expected that `pf.test("flow/flow.dag.yaml", inputs={"echo": "Hello"}, node="echo")` executes successfully with provided inputs.

**Screenshots**

**Running Information(please complete the following information):**
 - Promptflow Package Version using `pf -v`: 0.1.0b7.post1
 - Operating System: macOS 14.2.1 (23C71)
 - Python Version using `python --version`: Python 3.11.5

**Comment by 0mza987** [comment]
@bwilliams2 Thanks for you feedback, seems you are using a pretty old version of promptflow sdk, could you please try:
`pip install promptflow --upgrade` to update to the latest version `v1.3.0` and see if the bug still exists?

**Comment by bwilliams2** [comment]
You are correct. I observed the behavior in a newer version and replicated in a different environment. I just updated to the current version and have the same behavior.

Promptflow Package Version using pf -v: 
```
promptflow                       1.3.0

Executable '/Users/brycework/.pyenv/versions/3.11.5/bin/python3.11'
Python (Darwin) 3.11.5 (main, Sep 15 2023, 12:05:31) [Clang 14.0.3 (clang-1403.0.22.14.1)]
```

**Comment by 0mza987** [comment]
Add @lalala123123 to help take a look.

**Comment by lalala123123** [comment]
Hi @bwilliams2 , in promptflow test, the output of the linked node will be replaced with the corresponding value. Therefore, you need to specify the corresponding value of output in the inputs.

can you try to replace inputs with {"echo_object.output.echo": "hello"}?  It will work for you.
```
pf.test("flow/flow.dag.yaml", inputs={"echo_object.output.echo": "hello"}, node="echo")
```

**Comment by bwilliams2** [comment]
Hi @lalala123123, when I change to the suggested inputs I get the exact same error as shown before. I updated my test flow repository with the linked node reference.

```
File "/Users/brycework/.pyenv/versions/3.11.5/lib/python3.11/site-packages/promptflow/executor/_input_assignment_parser.py", line 105, in parse_node_property
    raise InvalidReferenceProperty(message_format=message_format, property=property, node_name=node_name) from e
promptflow.executor._errors.InvalidReferenceProperty
```

Additionally, when testing a node, I would not expect to have to define the linked node reference to a node input (`echo_object.output.echo`) rather than the named node input (`echo`) which is the approach in the promptflow documentation.

https://microsoft.github.io/promptflow/how-to-guides/init-and-test-a-flow.html#test-a-single-node-in-the-flow

**Comment by bwilliams2** [comment]
I looked at the new added test and that seems to be the same general configuration as my test case. I can successfully run the `test_pf_node_test_with_dict_input` test case but my example fails in the same environment. I will look deeper into any differences between the two.

**Comment by bwilliams2** [comment]
@lalala123123 I found the discrepancy between the two cases. The new test uses the full `get_dict_val` for the input to both of the `print_val` node inputs. dictionary input for both node inputs.

https://github.com/microsoft/promptflow/blob/e584700773fa72d239a7b70f127999cf9c1ae655/src/promptflow/tests/sdk_cli_test/e2etests/test_flow_test.py#L199-L202

The node output is type `dict` and has both `value` and `origin_value` keys which are provided to both input properties.

```
print(result.node_run_infos["get_dict_val"].output)
{'value': "{'input_key': 'input_value'}: <class 'dict'>", 'origin_value': {'input_key': 'input_value'}}
```

However, the bug I described uses string values for the node value inputs. The test case can be made equivalent to the initial bug report case with the changes below. With these modifications, the test case fails.

```
        inputs = {
            "get_dict_val.output.value": result.node_run_infos["get_dict_val"].output.value,
            "get_dict_val.output.origin_value": result.node_run_infos["get_dict_val"].output.origin_value,
        }
```
